Tests/SwiftFusionTests/Core/VectorTests.swift

```diff
@@ -1,283 +1,206 @@
-  /// `stride`.
-  fileprivate func makeVector(from start: Double, stride: Double) -> Testee {
-    return Testee(Array((0..<Self.dimension).map { start + Double($0) * stride }))
-  }
-
-  /// Tests all `FixedSizeVector` requirements.
-  func runAllFixedSizeVectorTests() {
-    testEquality()
-    testMove()
-    testAddMutating()
-    testAddFunctional()
-    testSubtractMutating()
-    testSubtractFunctional()
-    testScalarProductMutating()
-    testScalarProductFunctional()
-    testDot()
-    testNegate()
-    testSquaredNorm()
-    testNorm()
-    testDimension()
-    testWithUnsafeBufferPointer()
-    testWithUnsafeMutableBufferPointer()
-    testInitFromFlatTensor()
-    testFlatTensor()
-  }
-
-  /// Tests ==.
-  func testEquality() {
-    let v1 = makeVector(from: 0, stride: 0)
-    let v2 = makeVector(from: 1, stride: 0)
-    XCTAssertTrue(v1 == v1)
-    XCTAssertFalse(v1 == v2)
-  }
-
-  /// Tests move (exponential map).
-  func testMove() {
-    var v = makeVector(from: 0, stride: 1)
-    let t = makeVector(from: 1, stride: 1)
-    v.move(along: t)
-    XCTAssertEqual(v, makeVector(from: 1, stride: 2))
-  }
-
-  /// Tests the value and derivative of the mutating +=.
-  func testAddMutating() {
-    func doMutatingAdd(_ v1: Testee, _ v2: Testee) -> Testee {
-      var result = v1
-      result += v2
-      return result
-    }
-
-    let v1 = makeVector(from: 1, stride: 1)
-    let v2 = makeVector(from: 2, stride: 1)
-    XCTAssertEqual(doMutatingAdd(v1, v2), makeVector(from: 3, stride: 2))
-
-    let (value, pb) = valueWithPullback(at: v1, v2, in: doMutatingAdd)
-    XCTAssertEqual(value, makeVector(from: 3, stride: 2))
-    for v in basisVectors {
-      XCTAssertEqual(pb(v).0, v)
-      XCTAssertEqual(pb(v).1, v)
-    }
-  }
-
-  /// Tests the value and derivative of the functional +.
-  func testAddFunctional() {
-    let v1 = makeVector(from: 1, stride: 1)
-    let v2 = makeVector(from: 2, stride: 1)
-    XCTAssertEqual(v1 + v2, makeVector(from: 3, stride: 2))
-
-    let (value, pb) = valueWithPullback(at: v1, v2, in: +)
-    XCTAssertEqual(value, makeVector(from: 3, stride: 2))
-    for v in basisVectors {
-      XCTAssertEqual(pb(v).0, v)
-      XCTAssertEqual(pb(v).1, v)
-    }
-  }
-
-  /// Tests the value and derivative of the mutating -=.
-  func testSubtractMutating() {
-    func doMutatingSubtract(_ v1: Testee, _ v2: Testee) -> Testee {
-      var result = v1
-      result -= v2
-      return result
+extension Vector {
+  /// XCTests `self`'s semantic conformance to `Vector`, expecting its scalars to match
+  /// `expectedScalars`.
+  ///
+  /// Note: This does check the semantics of the vector math operations, but only using vectors
+  /// that have the same `scalar.indices` as `self`. If `Self` supports vector operations between
+  /// instances between different `scalars.indices`, then tests should additionally call
+  /// `checkVectorMathSemantics` with examples of vectors with differing `scalars.indices`.
```

Comment:
I don't like this restriction/distinction, especially in testing code where efficiency isn't a priority.  Can you explain why it's needed?